modin/pandas/test/dataframe/test_iter.py

```diff
@@ -257,6 +258,37 @@ def test___setattr__():
 
     df_equals(modin_df, pandas_df)
 
+    # While `new_col` is not a column of the dataframe,
+    # it should be accessible with __getattr__.
+    assert modin_df.new_col == pandas_df.new_col
+
+
+def test___setattr__mutating_column():
+    # Use case from issue #4577
+    pandas_df = pandas.DataFrame([[1]], columns=["col0"])
+    modin_df = pd.DataFrame([[1]], columns=["col0"])
+
+    # Replacing a column with a list should mutate the column in place.
```

Comment:
I think we should add a test here ensuring that we can do something like `ffill` on `df.col0` after setting it to a list

Reply:
Let's just check that `pandas_df.col0.equals(modin_df.col0)`. That will check that the types are `series`. That's what we care about checking

Reply:
I had that before, but got rid of it due to @mvashishtha 's suggestion. You can see it here: https://github.com/modin-project/modin/pull/4588#discussion_r902620705

Reply:
Makes sense!